env0/resource_environment.go

```diff
@@ -550,20 +550,66 @@ func createVariable(configurationVariable *client.ConfigurationVariable) any {
 }
 
 // Validate that the template is assigned to the "project_id".
-func validateTemplateProjectAssignment(d *schema.ResourceData, apiClient client.ApiClientInterface) error {
+func validateTemplateProjectAssignment(d *schema.ResourceData, template *client.Template) error {
 	projectId := d.Get("project_id").(string)
+
+	if projectId != template.ProjectId && !stringInSlice(projectId, template.ProjectIds) {
+		return errors.New("could not create environment: template is not assigned to project")
+	}
+
+	return nil
+}
+
+func createEnvironmentWithTemplate(d *schema.ResourceData, apiClient client.ApiClientInterface) (client.Environment, client.EnvironmentCreate, diag.Diagnostics) {
 	templateId := d.Get("template_id").(string)
 
 	template, err := apiClient.Template(templateId)
```

Comment:
taste `templateErr`

Reply:
The common/standard coding convention in golang is "err".
I will keep as is.